Sources/Datadog/RUM/RUMMonitor/Scopes/RUMUserActionScope.swift

```diff
@@ -83,51 +83,29 @@ internal class RUMUserActionScope: RUMScope {
 
         switch command {
         case let command as RUMStopUserActionCommand:
-            willStopUserAction(on: command)
+            sendActionEvent(completionTime: command.time, on: command)
             return false
 
         case let command as RUMResourceCommand:
             if command is  RUMStartResourceCommand {
-                startTrackingResource()
+                activeResourcesCount += 1
             } else if command is RUMStopResourceCommand {
-                stopTrackingResource()
+                activeResourcesCount -= 1
+                resourcesCount += 1
             } else if command is RUMStopResourceWithErrorCommand {
-                trackResourceError()
+                activeResourcesCount -= 1
+                errorsCount += 1
             }
 
         case _ as RUMAddCurrentViewErrorCommand:
-            trackViewError()
+            errorsCount += 1
```

Comment:
Are this counters thread safe?, can a user actions load some network resources that can fail asynchronously in different threads?

Reply:
Thread safety is ensured at the `RUMMonitor` level:
```swift
public class RUMMonitor {
    private let queue = DispatchQueue

    public methodExposedToUser(...) {
        process(...)
    }

    private func process(command: RUMCommand) {
        queue.async {
            _ = self.applicationScope.process(command: command)
        }
    }
}
```

Every processing after `RUMMonitor` (so in all `RUMScopes`) is single threaded.